### DISCUSSION TITLE
Mfkey32 nonce collection

### DISCUSSION THREAD

**Discussion by jekkos** [nfcgate/nfcgate#170] [Q&A] [answered]
Actually this is a generic question, not a bug report

**Device Information*** Which device are you using and what OS are you running on it?

 Google Pixel 3

**Log***

**Steps to reproduce**.

**Additional context**

I wonder if nfcgate would be able to capture nonces as it is required in mfkey32 attack. If I understand correctly then flipperzero is first partially emulating mifare classic, and then collecting the nonces that are then used to find at least one key. It would be great if I could grab this data from an nfc gate dump

**Proposed solution**

Potentially implement this feature, depending on technical feasibility

**Comment by kleest** [comment, marked as answer]
Hi @jekkos, you are correct that the Flipper Zero emulates a cloned Mifare Classic card or just generic ones against a reader device to collect nonces for a MFKey32 attack ([reference](https://docs.flipper.net/nfc/mfkey32)). Unfortunately NFCGate does not support emulating Mifare Classic because the underlying Android NFC-NCI stack does not. Therefore it's not possible to perform this attack with NFCGate.
